File: app/src/main/java/com/roy/turbo/launcher/helper/CheckLongPressHelper.java
package com.roy.turbo.launcher.helper;

import android.view.View;

import com.roy.turbo.launcher.LauncherAppState;

public class CheckLongPressHelper {
    private final View mView;
    private boolean mHasPerformedLongPress;
    private CheckForLongPress mPendingCheckForLongPress;

    class CheckForLongPress implements Runnable {
        public void run() {
            if ((mView.getParent() != null) && mView.hasWindowFocus()
                    && !mHasPerformedLongPress) {
                if (mView.performLongClick()) {
                    mView.setPressed(false);
                    mHasPerformedLongPress = true;
                }
            }
        }
    }

    public CheckLongPressHelper(View v) {
        mView = v;
    }

    public void postCheckForLongPress() {
        mHasPerformedLongPress = false;

        if (mPendingCheckForLongPress == null) {
            mPendingCheckForLongPress = new CheckForLongPress();
        }
        mView.postDelayed(mPendingCheckForLongPress,
                LauncherAppState.getInstance().getLongPressTimeout());
    }

    public void cancelLongPress() {
        mHasPerformedLongPress = false;
        if (mPendingCheckForLongPress != null) {
            mView.removeCallbacks(mPendingCheckForLongPress);
            mPendingCheckForLongPress = null;
        }
    }

    public boolean hasPerformedLongPress() {
        return mHasPerformedLongPress;
    }
}

File: app/src/main/java/com/roy/turbo/launcher/helper/CheckLongPressHelper.kt
package com.roy.turbo.launcher.helper

import android.view.View
import com.roy.turbo.launcher.LauncherAppState

class CheckLongPressHelper(private val mView: View) {
    private var mHasPerformedLongPress = false
    private var mPendingCheckForLongPress: CheckForLongPress? = null

    internal inner class CheckForLongPress : Runnable {
        override fun run() {
            if (mView.parent != null && mView.hasWindowFocus()
                && !mHasPerformedLongPress
            ) {
                if (mView.performLongClick()) {
                    mView.isPressed = false
                    mHasPerformedLongPress = true
                }
            }
        }
    }

    fun postCheckForLongPress() {
        mHasPerformedLongPress = false
        if (mPendingCheckForLongPress == null) {
            mPendingCheckForLongPress = CheckForLongPress()
        }
        mView.postDelayed(
            mPendingCheckForLongPress,
            LauncherAppState.getInstance().longPressTimeout.toLong()
        )
    }

    fun cancelLongPress() {
        mHasPerformedLongPress = false
        if (mPendingCheckForLongPress != null) {
            mView.removeCallbacks(mPendingCheckForLongPress)
            mPendingCheckForLongPress = null
        }
    }

    fun hasPerformedLongPress(): Boolean {
        return mHasPerformedLongPress
    }
}
